src/richie/apps/courses/api.py

```diff
@@ -45,3 +56,161 @@ def get_permissions(self):
                 permission_classes = self.permission_classes
 
         return [permission() for permission in permission_classes]
+
+
+# pylint: disable=too-many-return-statements,unused-argument, too-many-locals,too-many-branches
+@api_view(["POST"])
+def course_runs_sync(request, version):
+    """View for the web hook to create or update course runs based on their resource link.
+
+    - An existing course run is updated only if its "sync_mode" field is set to something else
+      than "manual".
+
+    - The public version of a course run is updated only if its "sync_mode" field is set to
+      "sync_to_public". Otherwise, only the draft version is updated and the related course
+      is marked dirty.
+
+    - A new course run is created only if the "DEFAULT_COURSE_RUN_SYNC_MODE" parameter is set
+      to something else than "manual" in the lms configuration (or the setting
+      "RICHIE_DEFAULT_COURSE_RUN_SYNC_MODE" in the absence of LMS preference). Otherwise, only
+      existing course runs are updated.
+
+    - A new public course run is created only if the "DEFAULT_COURSE_RUN_SYNC_MODE" parameter
+      is set to "sync_to_public" in the lms configuration (or the setting
+      "RICHIE_DEFAULT_COURSE_RUN_SYNC_MODE" in the absence of LMS preference). Otherwise, only
+      the draft course run is created and the related course is marked dirty.
+
+    Parameters
+    ----------
+    request : Type[django.http.request.HttpRequest]
+        The request on the API endpoint, it should contain a payload with course run fields.
+
+    Returns
+    -------
+    Type[rest_framework.response.Response]
+        HttpResponse acknowledging the success or failure of the synchronization operation.
+    """
+    msg = request.body.decode("utf-8")
+
+    # Check if the provided signature is valid against any secret in our list
+    #
+    # We need to do this to support 2 or more versions of our infrastructure at the same time.
+    # It then enables us to do updates and change the secret without incurring downtime.
+    authorization_header = request.headers.get("Authorization")
+    if not authorization_header:
+        return Response("Missing authentication.", status=401)
+
+    signature_is_valid = any(
+        authorization_header
```

Comment:
In fact the `Authorization` header should not contains directly the `hash` but also the authorization type. it should with this pattern: `Authorization: <type> <value>`

Reply:
Good point.  So what should the type be. My own?

Reply:
Sorry, it reminds me this : https://www.youtube.com/watch?v=Ki_2rVqz_Qc 😂

But maybe the type should describe what is the hash ? Like aws signature ?

Reply:
I can put `YOUROWN` if you insist or `SIGN-HMAC-SHA256` to follow AWS?

Reply:
Yes `SIG-HMAC-SHA256` is perfect